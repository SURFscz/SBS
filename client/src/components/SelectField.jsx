import React from "react";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import ReactTooltip from "react-tooltip";
import "./SelectField.scss";
import Select from "react-select";

export default function SelectField({
                                        onChange, name, value, options, placeholder = "", disabled = false,
                                        toolTip = null, searchable = false, className = "input-select-inner",
                                        classNamePrefix = "select-inner", clearable = false
                                    }) {
    return (
        <div className="select-field">
            <label htmlFor={name}>{name} {toolTip &&
            <span className="tool-tip-section">
                <span data-tip data-for={name}><FontAwesomeIcon icon="info-circle"/></span>
                <ReactTooltip id={name} type="light" effect="solid" data-html={true}>
                    <p dangerouslySetInnerHTML={{__html: toolTip}}/>
                </ReactTooltip>
            </span>}
            </label>
            <Select
                className={className}
                classNamePrefix={classNamePrefix}
                value={value}
                placeholder={placeholder}
                isDisabled={disabled}
                onChange={onChange}
                options={options}
                isSearchable={searchable}
                isClearable={clearable}
            />
        </div>
    );
}
