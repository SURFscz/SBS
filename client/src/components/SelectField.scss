@import "../stylesheets/vars.scss";

.select-field {
    display: flex;
    align-items: center;

    label {
        display: inline-block;
        padding: 14px 34px 14px 14px;
        margin: 5px 0;
        min-width: 180px;
        border: 1px solid $lighter-grey;
        background-color: $lightest-grey;
        border-bottom-left-radius: $br;
        border-top-left-radius: $br;
        color: $dark-green;
        position: relative;

    }

    span.tool-tip-section {
        position: absolute;
        right: 0;
        top: 0;
        font-size: 14px;
        color: $blue;
        cursor: pointer;
        padding: 0 4px;
    }

    .input-select-inner {
        display: block;
        width: 100%;
        border-bottom: 1px solid $lighter-grey;
        border-right: 1px solid $lighter-grey;
        border-top: 1px solid $lighter-grey;
        border-left: none;
        border-bottom-right-radius: $br;
        border-top-right-radius: $br;
        font-size: 14px;
        height: 50px;

        .select-inner__control {
            height: 100%;
            padding-left: 8px;
            border: none;
        }

        .select-inner__control--is-focused {
            box-shadow: none !important;
            border: #5f9af6 2px solid !important;
            outline-offset: -2px;
            outline-color: #5f9af6 !important;
            border-radius: 0;

        }

        &[disabled] {
            background-color: $background;
            cursor: not-allowed;
        }

    }

    .select-inner__control--is-disabled {
        background-color: $background;
        border-radius: 0;
    }
}