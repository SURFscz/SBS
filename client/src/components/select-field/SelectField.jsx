import React from "react";

import "./SelectField.scss";
import Select from "react-select";
import CreatableSelect from "react-select/creatable";
import {Tooltip} from "@surfnet/sds";
import ClipBoardCopy from "../_redesign/ClipBoardCopy";
import {isEmpty} from "../../utils/Utils";

export default function SelectField({
                                        onChange,
                                        name,
                                        value,
                                        options,
                                        placeholder = "",
                                        disabled = false,
                                        toolTip = null,
                                        searchable = false,
                                        small = false,
                                        clearable = false,
                                        isMulti = false,
                                        creatable = false,
                                        onInputChange = null,
                                        copyClipBoard = false,
                                        isOptionDisabled = () => false,
                                        required = false,
                                        error = false
                                    }) {
    const styles = {
        multiValue: (base, state) => state.data.isFixed ? {
            ...base,
            backgroundColor: 'var(--sds--color--gray--200)!important'
        } : base,
        multiValueRemove: (base, state) => state.data.isFixed ? {...base, display: 'none'} : base
    };
    return (
        <div className="select-field">
            <label htmlFor={name}>{name}{required && <sup className="required">*</sup>}
                {toolTip && <Tooltip tip={toolTip}/>}
            </label>
            <div className="select-field-inner">
                {creatable &&
                    <CreatableSelect
                        className={`input-select-inner creatable ${error ? "error" : ""}`}
                        classNamePrefix={"select-inner"}
                        value={value}
                        isMulti={true}
                        placeholder={placeholder}
                        isSearchable={true}
                        onInputChange={onInputChange}
                        isClearable={clearable}
                        isDisabled={disabled}
                        onChange={onChange}
                        options={options}
                    />}
                {!creatable &&
                    <Select
                        className={`input-select-inner ${small ? " small" : ""}  ${error ? "error" : ""}`}
                        classNamePrefix={"select-inner"}
                        value={value}
                        placeholder={placeholder}
                        isDisabled={disabled}
                        onChange={onChange}
                        styles={isMulti ? styles : undefined}
                        isMulti={isMulti}
                        options={options}
                        isSearchable={searchable}
                        isClearable={clearable}
                        isOptionDisabled={isOptionDisabled}
                    />}
                {copyClipBoard && <ClipBoardCopy
                    txt={isEmpty(value) ? "" : Array.isArray(value) ? value.map(v => v.label).join(", ") : value}
                    right={true}/>}
            </div>
        </div>
    );
}
