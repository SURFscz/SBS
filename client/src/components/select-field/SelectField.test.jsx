import React from "react";
import {render} from "@testing-library/react";
import SelectField from "./SelectField";

describe("SelectField", () => {
    it("matches snapshot with minimal props", () => {
        const {asFragment} = render(
            <SelectField
                name="Test"
                value="Value"
                onChange={() => {}}
                options={[]}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("matches snapshot with options", () => {
        const {asFragment} = render(
            <SelectField
                name="Test"
                value="Value"
                onChange={() => {}}
                options={[{value: "Option1", label: "Option 1"}]}
            />
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
