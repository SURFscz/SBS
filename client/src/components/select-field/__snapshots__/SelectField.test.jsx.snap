// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectField matches snapshot with minimal props 1`] = `
<DocumentFragment>
  <div
    class="select-field"
  >
    <label
      for="Test"
    >
      Test
    </label>
    <div
      class="select-field-inner"
    >
      <div
        class="input-select-inner css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
          role="log"
        />
        <div
          class="select-inner__control css-13cymwt-control"
        >
          <div
            class="select-inner__value-container css-hlgwow"
          >
            <div
              class="select-inner__placeholder css-1jqq78o-placeholder"
              id="react-select-2-placeholder"
            />
            <input
              aria-activedescendant=""
              aria-autocomplete="list"
              aria-describedby="react-select-2-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              aria-readonly="true"
              class="css-1s80ejz-dummyInput-DummyInput"
              id="react-select-2-input"
              inputmode="none"
              role="combobox"
              tabindex="0"
              value=""
            />
          </div>
          <div
            class="select-inner__indicators css-1wy0on6"
          >
            <span
              class="select-inner__indicator-separator css-1u9des2-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="select-inner__indicator select-inner__dropdown-indicator css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SelectField matches snapshot with options 1`] = `
<DocumentFragment>
  <div
    class="select-field"
  >
    <label
      for="Test"
    >
      Test
    </label>
    <div
      class="select-field-inner"
    >
      <div
        class="input-select-inner css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-3-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
          role="log"
        />
        <div
          class="select-inner__control css-13cymwt-control"
        >
          <div
            class="select-inner__value-container css-hlgwow"
          >
            <div
              class="select-inner__placeholder css-1jqq78o-placeholder"
              id="react-select-3-placeholder"
            />
            <input
              aria-activedescendant=""
              aria-autocomplete="list"
              aria-describedby="react-select-3-placeholder"
              aria-expanded="false"
              aria-haspopup="true"
              aria-readonly="true"
              class="css-1s80ejz-dummyInput-DummyInput"
              id="react-select-3-input"
              inputmode="none"
              role="combobox"
              tabindex="0"
              value=""
            />
          </div>
          <div
            class="select-inner__indicators css-1wy0on6"
          >
            <span
              class="select-inner__indicator-separator css-1u9des2-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class="select-inner__indicator select-inner__dropdown-indicator css-1xc3v61-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
